benchmark/index.js

```diff
@@ -0,0 +1,203 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable func-names */
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable guard-for-in */
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-use-before-define */
+/* eslint-disable import/no-dynamic-require */
+/* eslint-disable global-require */
+const Benchmark = require("benchmark-suite");
+const fs = require("fs");
+const path = require("path");
+const Commander = require("commander");
+const logger = require("./metaLogger");
+const versionedRouter = require("../versionedRouter");
+const cdkV2Handler = require("../cdk/v2/handler");
+const { TRANSFORMER_METRIC } = require("../v0/util/constant");
+
+const supportedDestinations = ["algolia", "pinterest_tag"];
+
+logger.info();
+
+const command = new Commander.Command();
+command
+  .allowUnknownOption()
+  .option(
+    "-d, --destinations <string>",
+    "Enter destination names separated by comma",
+    supportedDestinations.toString()
+  )
+  .option(
+    "-bt, --benchmarktype <string>",
+    "Enter the benchmark type (Operations or Memory)",
+    "Operations"
+  )
+  .parse();
+
+const testDataDir = path.join(__dirname, "./../__tests__/data");
+const getTestData = (intgList, fileNameSuffixes) => {
+  const intgTestData = {};
+  intgList.forEach(intg => {
+    // Use the last valid test data file
+    fileNameSuffixes.forEach(fileNameSuffix => {
+      try {
+        intgTestData[intg] = JSON.parse(
+          fs.readFileSync(
+            path.join(testDataDir, `${intg}${fileNameSuffix}.json`),
+            {
+              encoding: "utf-8"
+            }
+          )
+        );
+      } catch (err) {
+        // logger.error(
+        //   `Unable to load the data for: "${intg}" suffix: "${fileNameSuffix}"`
+        // );
+        // logger.error(`Raw error: "${err}"`);
+      }
+    });
+  });
+  return intgTestData;
+};
+
+const cmdOpts = command.opts();
+
+// Initialize data for destinations
+const destinationsList = cmdOpts.destinations
+  .split(",")
+  .map(x => x.trim())
+  .filter(x => x !== "");
+logger.info("Destinations selected: ", destinationsList);
+logger.info();
+const destDataset = getTestData(destinationsList, ["_input", ""]);
+
+const nativeDestHandlers = {};
+const destCdKWorkflowEngines = {};
+
+const benchmarkType = cmdOpts.benchmarktype.trim();
+
+async function initializeHandlers() {
+  for (const idx in destinationsList) {
+    const dest = destinationsList[idx];
+
+    // Native destination handler
+    nativeDestHandlers[dest] = versionedRouter.getDestHandler(
+      "v0",
+      dest
+    ).process;
```

Comment:
what about router transformation flow benchmarking ? `processRouterDest`

Reply:
At this point, we are only supporting processor transformation. The router support will be added as a next step.